### PR TITLE
Added Palindrome in Kotlin

### PR DESCRIPTION
**Pull request by anohene1**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][1]! 
For simplicity, please make sure that your pull request includes one and only one contribution.

## Complete the Applicable Sections Below

Find which section best describes your pull request and make sure you fill it out. To start, let us know which issue you've fixed.

- [ ] I fixed #your-issue-number-here

### Code Snippets

- [x] I named the pull request using `Added/Updated <Sample Program> in <Language>` format
- [ ] I created/updated the language README
  - [ ] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])

### Testing

- [x] I named the pull request using `Added/Updated <Language>/<Project> Testing` format
- [x] I followed the testinfo template, if applicable

## Notes

Feel free to describe what you added or updated.

[1]: https://therenegadecoder.com/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism

### CODE REVIEW

**Review by jrg94 — CHANGES_REQUESTED** [review]
We don't currently support palindrome, but we're open to including this if we can get it documented. Let me know if that's something you want to do!

**Comment by anohene1** [comment]
Sure. I'll do that. Let me know where to document it.

**Comment by jrg94** [comment]
Check out this repo: https://github.com/TheRenegadeCoder/sample-programs-website

You'll want to create something that looks like one of these: https://sample-programs.therenegadecoder.com/projects/

**Comment by anohene1** [comment]
I have done that and made a pull request. You can check it at https://github.com/TheRenegadeCoder/sample-programs-website/pull/349

**Comment by jrg94** [comment]
Cool! Now we have the specification, let's get some tests going. There's a bit of a dependency in our way at the moment. See #2186. That said, we can add the test file here: https://github.com/TheRenegadeCoder/sample-programs/tree/master/test

**Comment by anohene1** [comment]
I'm sorry I don't get you quite well. Should the test be in python? As I can see all the test files are written in python.

**Comment by jrg94** [comment]
Yes! But the test files are more or less copy-and-paste. You should be able to copy binary-search.py and reuse most of it.

**Comment by anohene1** [comment]
Okay I'll go ahead and try that. I'll let you know if I'm done.

**Comment by anohene1** [comment]
This is getting very difficult for me as I have no knowledge in python.

**Comment by jrg94** [comment]
No worries! I can give you a rough idea of what you need to do. First, I'd copy the contents of the binary search test.

**Comment by jrg94** [comment]
Then, replace BinarySearch with PalindromicNumbers.

**Comment by jrg94** [comment]
Then, at the top of the file, you should see these: 

```
invalid_permutations = (
    'description,in_params,expected', [
        (
            'no input',
            None,
            'Usage: please provide a list of sorted integers ("1, 4, 5, 11, 12") and the integer to find ("11")'
        ), (
            'missing input: target',
            '"1, 2, 3, 4"',
            'Usage: please provide a list of sorted integers ("1, 4, 5, 11, 12") and the integer to find ("11")'
        ), (
            'missing input: list',
            '"" "5"',
            'Usage: please provide a list of sorted integers ("1, 4, 5, 11, 12") and the integer to find ("11")'
        ), (
            'out of order input',
            '"3, 5, 1, 2" "3"',
            'Usage: please provide a list of sorted integers ("1, 4, 5, 11, 12") and the integer to find ("11")'
        )
    ]
)

valid_permutations = (
    'description,in_params,expected', [
        (
            'sample input first true',
            '"1, 3, 5, 7" "1"',
            'true'
        ), (
            'sample input last true',
            '"1, 3, 5, 7" "7"',
            'true'
        ), (
            'sample input middle true',
            '"1, 3, 5, 7" "5"',
            'true'
        ), (
            'sample input one true',
            '"5" "5"',
            'true'
        ), (
            'sample input one false',
            '"5" "7"',
            'false'
        ), (
            'sample input many false',
            '"1, 3, 5, 6" "7"',
            'false'
        )
    ]
)
```

**Comment by jrg94** [comment]
These are the tests. You need to replace these tests with the one's you wrote. For example, the first invalid test looks like this: 

```
(
            'no input',
            None,
            'Usage: please provide a list of sorted integers ("1, 4, 5, 11, 12") and the integer to find ("11")'
)
```

You can replace it with this:

```
(
            'no input',
            None,
            'Usage: please input a number'
)
```

**Comment by anohene1** [comment]
I have finished editing the test file. You may check and let me know what else I need to do please.

File: test/projects/test_palindromic_number.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+invalid_permutations = (
+    'description,in_params,expected', [
+        (
+            'no input',
+            None,
+            'Usage: please input a number'
+        ), (
+            'empty input',
+            None,
```
Comment (jrg94) [inline test/projects/test_palindromic_number.py:15]:
```suggestion
            "",
```

File: test/projects/test_palindromic_number.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            'Usage: please input a number'
+        ), (
+            'invalid input: not a number',
+            None,
```
Comment (jrg94) [inline test/projects/test_palindromic_number.py:19]:
```suggestion
            "a",
```

File: test/projects/test_palindromic_number.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+def palindromic_number(request):
+    request.param.build()
+    yield request.param
+    request.param.cleanup()
+
+
+@project_test(ProjectType.PalindromicNumber.key)
+@pytest.mark.parametrize(valid_permutations[0], valid_permutations[1],
+                         ids=[p[0] for p in valid_permutations[1]])
+def test_palindromic_number_valid(description, in_params, expected, binary_search):
```
Comment (jrg94) [inline test/projects/test_palindromic_number.py:50]:
```suggestion
def test_palindromic_number_valid(description, in_params, expected, palindromic_number):
```

File: test/projects/test_palindromic_number.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    actual = palindromic_number.run(params=in_params)
+    assert actual.strip().lower() == expected
+
+
+@project_test(ProjectType.PalindromicNumber.key)
+@pytest.mark.parametrize(invalid_permutations[0], invalid_permutations[1],
+                         ids=[p[0] for p in invalid_permutations[1]])
+def test_palindromic_number_invalid(description, in_params, expected, binary_search):
```
Comment (jrg94) [inline test/projects/test_palindromic_number.py:58]:
```suggestion
def test_palindromic_number_invalid(description, in_params, expected, palindromic_number):
```

**Review by jrg94 — CHANGES_REQUESTED** [review]
Looks good so far!

**Comment by jrg94** [comment]
After this, you'll need to update both of these:

- https://github.com/TheRenegadeCoder/sample-programs/blob/master/runner.py
- https://github.com/TheRenegadeCoder/sample-programs/blob/master/.glotter.yml

**Comment by anohene1** [comment]
I have made both updates

**Review by jrg94 — CHANGES_REQUESTED** [review]
Looks good! The only issue is the program you wrote will fail your own tests. We don't have a way to prove that just yet because someone is currently adding Kotlin testing.

**Comment by anohene1** [comment]
Okay. So what do I do now?

**Comment by jrg94** [comment]
Update the Kotlin script to match your tests. Then, we're stuck waiting on #2186.

**Comment by anohene1** [comment]
I have fixed that.

**Review by jrg94 — APPROVED** [review]
I'm not sure your program matches your tests, but we'll go ahead and merge it for now. When Kotlin testing is implemented, we'll clean things up.

**Pull request merged**